test/WebJobs.Script.Tests/ScriptHostTests.cs

```diff
@@ -1488,6 +1488,53 @@ public void IsFunction_ReturnsExpectedResult()
             Assert.False(host.IsFunction(null));
         }
 
+        [Fact]
+        public void Initialize_LogsWarningForExplicitlySetHostId()
+        {
+            TestLoggerProvider loggerProvider = null;
+            var loggerFactoryHookMock = new Mock<ILoggerProviderFactory>(MockBehavior.Strict);
+            loggerFactoryHookMock
+                .Setup(m => m.CreateLoggerProviders(It.IsAny<ScriptHostConfiguration>(), It.IsAny<ScriptSettingsManager>(), It.IsAny<Func<bool>>(), It.IsAny<Func<bool>>()))
+                .Returns<ScriptHostConfiguration, ScriptSettingsManager, Func<bool>, Func<bool>>((scriptConfig, settingsManager, fileLoggingEnabled, isPrimary) =>
+                {
+                    loggerProvider = new TestLoggerProvider(scriptConfig.LogFilter.Filter);
+                    return new[] { loggerProvider };
+                });
```

Comment:
Since you're not doing anything to verify the guts of the `LoggerFactory`, it's easier to just use a regular `LoggerFactory` here without mocking. Just add your `TestLoggerProvider` to it, and then pass that `LoggerFactory` into the `ScriptHost` constructor.